app/src/main/java/com/expidev/gcmapp/support/v4/content/CurrentAssignmentLoader.java

```diff
@@ -0,0 +1,182 @@
+package com.expidev.gcmapp.support.v4.content;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+
+import com.expidev.gcmapp.db.Contract;
+import com.expidev.gcmapp.db.MinistriesDao;
+import com.expidev.gcmapp.model.Assignment;
+import com.expidev.gcmapp.utils.BroadcastUtils;
+
+import org.ccci.gto.android.common.support.v4.content.AsyncTaskBroadcastReceiverSharedPreferencesChangeLoader;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.expidev.gcmapp.Constants.PREFS_SETTINGS;
+import static com.expidev.gcmapp.Constants.PREF_CURRENT_ASSIGNMENT;
+import static com.expidev.gcmapp.Constants.PREF_CURRENT_MINISTRY;
+
+/**
+ * Created by William.Randall on 2/25/2015.
+ */
+public class CurrentAssignmentLoader extends AsyncTaskBroadcastReceiverSharedPreferencesChangeLoader<Assignment>
```

Comment:
This entire Loader will probably be replaced by the one in #50. I will probably pull your role based selection logic though for the initial selection.